conans/client/generators/b2.py

```diff
@@ -279,24 +279,25 @@ def b2_variation(self):
         return self._b2_variation
 
     @property
-    def b2_toolset_name(self):
+    def b2_toolset(self):
         compiler = {
             'sun-cc': 'sun',
             'gcc': 'gcc',
             'Visual Studio': 'msvc',
             'clang': 'clang',
             'apple-clang': 'clang'
         }.get(self.conanfile.settings.get_safe('compiler'))
-        return str(compiler)
+        if not compiler:
+            return
```

Comment:
otherwise what? Looks like a bug.

Reply:
Otherwise it constructs a toolset name from `compiler` and compiler version. If explicit `else` is preferred, I can add it.

Reply:
oh! sorry, the method now continues. It is ok.